### PR TITLE
$newRating is not an object or entity

### PR DESCRIPTION
**Pull request by saeideng**

### CODE REVIEW

**Comment by dereuromark** [comment]
maybe it is and doc block is wrong?

**Comment by saeideng** [comment]
Notice (8): Trying to get property 'newRating' of non-object [ROOT/vendor/dereuromark/cakephp-ratings/src/Controller/Component/RatingComponent.php, line 111]

**Comment by saeideng** [comment]
it is float I think

**Comment by dereuromark** [comment]
Tests are failing
It looks like it can be both float and entity depending on which case.
We might want to either fix the whole API here to always return entities, or we need to fix the docblock to state float|Rating
Float itself seems worse: https://github.com/dereuromark/cakephp-ratings/compare/master...dereuromark:try-float

**Comment by saeideng** [comment]
just one of `float` or `entity` acceptable IMO
I will review the plugin again

**Comment by saeideng** [comment]
> It looks like it can be both float and entity depending on which case.

yes 
here returns float
https://github.com/dereuromark/cakephp-ratings/blob/d85e4ce709f36b85b42b4c5aa16d6416b034aaa9/src/Model/Behavior/RatableBehavior.php#L374

and here entity
https://github.com/dereuromark/cakephp-ratings/blob/d85e4ce709f36b85b42b4c5aa16d6416b034aaa9/src/Model/Behavior/RatableBehavior.php#L385-L387

depends to people use/have `rating` field in their DB or no :)

**Comment by saeideng** [comment]
it can not return entity all cases , so I think it should return just float

**Comment by saeideng** [comment]
>We might want to either fix the whole API here to always return entities

which entity ? entity of Ratings table or user table (Article table for example )

**Comment by saeideng** [comment]
```php
$Rating->newRating
$Rating->oldRating
```
are in cake 2 style that used in this plugin , they are in table but cake returns them with entity

**Comment by dereuromark** [comment]
Good question. I am also fine with your solution.

**Comment by saeideng** [comment]
I checked again whole the behavior
there are 2 methods
```php
saveRating()
removeRating()
```
that I think those can return entities always 
and some other methods like 
```php
incrementRating()
calculateRating()
decrementRating()
```
these returns entity when `saveToField` is `true` or a string , otherwise return a float
I dont know we able to change method's signature for these 
to return always entity

also when we expect to result from these methods(like `calculateRating`)  without saving in DB 
returning entity ( dirty entity ) I think is pointless

**Comment by dereuromark** [comment]
what are we doing with this one?

**Comment by dereuromark** [comment]
ping @saeideng

File: src/Controller/Component/RatingComponent.php [hunk 1]
```diff
@@ -108,7 +108,7 @@ public function rate($rate, $rating, $user, $redirect = false) {
 			$Model = $Controller->{$this->getConfig('modelName')};
 			$newRating = $Model->saveRating($rate, $user, $rating);
 			if ($newRating) {
-				$rating = round($newRating->newRating);
+				$rating = round($newRating);
```
Comment (dereuromark) [inline src/Controller/Component/RatingComponent.php:111]: I changed it to $newRating->rating, as it is apparently an entity now.
See master code.

**Pull request closed without merging**